ledger-core/virtual/integration/method_test.go

```diff
@@ -1191,3 +1192,201 @@ func TestVirtual_CallMultipleContractsFromContract_Ordered(t *testing.T) {
+		outgoingReasonSecond = reference.NewSelf(outgoingSecondCall)
+	)
+
+	// create objects
+	{
+		Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+		Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+	}
+
+	// add ExecutionMocks to runnerMock
+	{
+		firstBuilder := execution.NewRPCBuilder(outgoingReasonFirst, objectAGlobal)
+		objectAExecutionFirstMock := runnerMock.AddExecutionMock(outgoingFirstCall.String())
+		objectAExecutionFirstMock.AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [A.Foo] first call")
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Outgoing: firstBuilder.CallMethod(objectBGlobal, classB, "Bar", []byte("first")),
+			},
+		)
+		objectAExecutionFirstMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.Foo] first call")
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Outgoing: firstBuilder.CallMethod(objectBGlobal, classB, "Bar", []byte("second")),
+			},
+		)
+		objectAExecutionFirstMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.Foo] first call")
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+			},
+		)
+
+		secondBuilder := execution.NewRPCBuilder(outgoingReasonSecond, objectAGlobal)
+		objectAExecutionSecondMock := runnerMock.AddExecutionMock(outgoingSecondCall.String())
+		objectAExecutionSecondMock.AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [A.Foo] second call")
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Outgoing: secondBuilder.CallMethod(objectBGlobal, classB, "Bar", []byte("first")),
+			},
+		)
+		objectAExecutionSecondMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.Foo] second call")
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Outgoing: secondBuilder.CallMethod(objectBGlobal, classB, "Bar", []byte("second")),
+			},
+		)
+		objectAExecutionSecondMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.Foo] second call")
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify(outgoingFirstCall.String(), flags, nil)
+		runnerMock.AddExecutionClassify(outgoingSecondCall.String(), flags, nil)
+	}
+
+	// add publish checker
+	{
+		server.PublisherMock.SetChecker(func(topic string, messages ...*message.Message) error {
+			require.Len(t, messages, 1)
+
+			pl, err := payload.UnmarshalFromMeta(messages[0].Payload)
+			require.NoError(t, err)
+
+			switch pl.(type) {
+			case *payload.VCallRequest:
```

Comment:
server.PublisherMock.SetTypedChecker

Reply:
I don't want to resend msg to B, I need to make VCallResult by myself. 
If I'm right, it's impossible with TypedChecker.

Reply:
it's possible and works just find

Reply:
done